### PR TITLE
Refactor fxprequests

### PR DESCRIPTION
**Pull request by ramirogm**
In this PR I'm doing two things ( which is probably not such a good idea ):
- Refactor the fxp-related changes to be as independent as possible from upstream, in order to make pulling changes from there easier
- Pulling upstream changes from `mojaloop/mojaloop-sdk-scheme-adapter`

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -37,7 +42,7 @@ let config = {
     outboundPort: 4001,
     peerEndpoint: 'peerEndpoint:3001',
     backendEndpoint: 'backendEndpoint:3001',
-    forwardPutTransfersToBackend: true,
+    fxpModeEnabled: false,
```
Comment (ramirogm) [inline src/config.js:45]: Renaming the flag

File: src/config.js [hunk 2]
```diff
@@ -63,7 +68,8 @@ let config = {
     cacheConfig: {
         host: 'localhost',
         port: 6379
-    }
+    },
+    enableTestFeatures: false
```
Comment (ramirogm) [inline src/config.js:74]: from upstream

File: src/inboundApi/api.yaml [hunk 3]
```diff
@@ -2498,8 +2498,39 @@ paths:
           $ref: '#/components/responses/ErrorResponse503'
       requestBody:
         $ref: '#/components/requestBodies/ErrorInformationObject'
-servers:
-  - url: /fsp
+
```
Comment (ramirogm) [inline src/inboundApi/api.yaml:2501]: from upstream

**Comment by ramirogm** [comment]
@chinogarcia Don't review yet. It's not ready

**Comment by ramirogm** [comment]
@chinogarcia Ready to be reviewed

File: src/inboundApi/fxpHandlers.js [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+};
+
+
+/**
+ * Handles a POST /quotes request
+ */
+const postQuotes = async (ctx) => {
+    if (!ctx.fxpQuote) {
+        ctx.response.status = 400;
+        ctx.response.body = 'Can not handle non FXP quotes'; // FIXME
```
Comment (ramirogm) [inline src/inboundApi/fxpHandlers.js]: FIXME what???

File: src/lib/model/fxpInboundModel.js [hunk 5]
```diff
@@ -0,0 +1,611 @@
...
+            this.logger.log(`Ignoring cache notification for transfer ${secondStageQuoteId}. Type is not quoteResponse: ${util.inspect(message)}`);
+            return;
+        }
+        const secondStageQuoteResponse = message.data;
+        const secondStageQuoteResponseHeaders = message.headers;
+
+        // cancel the timeout handler
+        // clearTimeout(timeout); // FIXME implement timeouts
+
+        // clean quoteId from response if there. The SDK returns it as part of a quote response but it's not part of it per the Mojaloop spec ( FIXME find proper reference )
```
Comment (ramirogm) [inline src/lib/model/fxpInboundModel.js]: Well, a previous version of the SDK

File: src/lib/model/fxpInboundModel.js [hunk 6]
```diff
@@ -0,0 +1,611 @@
...
+        }
+        catch (error) {
+            this.logger.log(`Error from fxp to secondStageQuoteResponse. Making error callback to ${originalQuoteSourceFspId}`);
+            const err = new Errors.MojaloopFSPIOPError(error, error.message, originalQuoteSourceFspId, Errors.MojaloopApiErrorCodes.PAYEE_ERROR);
+            // FIXME wrap in a trycatch and log
+            return await this.mojaloopRequests.putQuotesError(originalQuoteRequest.quoteId, err.toApiErrorObject(), originalQuoteSourceFspId);
+        }
+        const sourceFspId = composedResponseToOriginalQuote.metadata.sourceFSP;
+        const destinationFspId = composedResponseToOriginalQuote.metadata.destinationFSP;
+        this.logger.log(`[Quotes 30] FXP QUOTE : creating quote response IlpPacker, condition and fulfilment on ${util.inspect(composedResponseToOriginalQuote)}`);
```
Comment (ramirogm) [inline src/lib/model/fxpInboundModel.js]: IlpPackeT not packeR

File: src/lib/model/fxpInboundModel.js [hunk 7]
```diff
@@ -0,0 +1,611 @@
...
+
+        // create our ILP packet and condition and tag them on to our internal quote response 
+        const { fulfilment, ilpPacket, condition } = this.ilp.getQuoteResponseIlp(secondStageQuote, responseToOriginalQuote);
+
+        responseToOriginalQuote.ilpPacket = ilpPacket;
+        responseToOriginalQuote.condition = condition; 
+
+        // now store the fulfilment and the quote data against the quoteId in our cache
+        // as we are going to use this on the transfer processing
+        await this.cache.set(`quote_${originalQuoteRequest.transactionId}`, {
```
Comment (ramirogm) [inline src/lib/model/fxpInboundModel.js:240]: Maybe use transaction_ as the prefix? instead of quote_

File: src/lib/model/fxpInboundModel.js [hunk 8]
```diff
@@ -0,0 +1,611 @@
...
+            originalQuoteSourceFspId,
+            originalQuoteDestinationFspId,
+            secondStageQuoteSourceFsp,
+            secondStageQuoteDestinationFsp,
+            originalQuoteRequest,
+            responseToOriginalQuote,
+            secondStageQuote,
+            secondStageQuoteResponse,
+            mojaloopResponse: responseToOriginalQuote,
```
Comment (ramirogm) [inline src/lib/model/fxpInboundModel.js]: This alias is not needed in fxpModel

**Review by ramirogm — COMMENTED** [review]
See comments

**Review by chinogarcia — APPROVED** [review]

**Pull request merged**